search-commons/src/test/java/no/unit/nva/search/IndexDocumentTest.java

```diff
@@ -70,21 +85,59 @@ public void toIndexDocumentReturnsIndexDocumentWitEmptyContributorListWhenPublic
     @Test
     public void toIndexDocumentReturnsIndexDocumentWithNoDateWithPublicationDateIsNull()
         throws MalformedURLException, InvalidIssnException {
-        Publication publication = PublicationGenerator.publicationWithIdentifier();
+        Publication publication = publicationWithIdentifier();
         publication.getEntityDescription().setDate(null);
         IndexDocument indexDocument = IndexDocument.fromPublication(publication);
         assertThat(indexDocument.getPublicationDate(),is(nullValue()));
     }
 
 
+    @Test
+    void toJsonStringSerializesUrisAsIdentifiers()
+            throws InvalidIsbnException, InvalidUnconfirmedSeriesException {
+
+        final URI bookSeriesId = randomPublicationChannelsUri();
+        final URI publisherId = randomPublicationChannelsUri();
+        final PublishingHouse publisher = publishingHouseWithUri(publisherId);
+        EntityDescription entityDescription = createSampleEntityDescriptionBook(bookSeriesId, publisher);
+        Publication publication = createPublicationWithEntityDescription(entityDescription);
+        assertThat(publication, doesNotHaveEmptyValuesIgnoringFields(IGNORED_PUBLICATION_FIELDS));
+        IndexDocument actualDocument = IndexDocument.fromPublication(publication);
+        assertThat(actualDocument, doesNotHaveEmptyValuesIgnoringFields(IGNORED_INDEXED_DOCUMENT_FIELDS));
+    }
+
+    @Test
+    public void toJsonPreservesRegistredDataWhenIdentifierIsUriFromPublicationChannels() throws Exception {
```

Comment:
There is not serialization here, only converting to IndexDocument

Reply:
Renamed methods to resemble behavior